.github/copilot-instructions.md

```diff
@@ -0,0 +1,188 @@
+        Card.stories.tsx
+      ...
+    index.ts              # Root barrel: export * from './ui';
+  lib/
+    utils.ts              # cn() and other generic helpers
+  hooks/                  # Shared hooks (if any)
+```
+
+## Component Authoring Pattern
+- File name matches component name (PascalCase).
+- Default export: the main React component (or named export if multiple primitives).
+- Provide a `Props` interface and export it.
+- Support `className` passthrough merged via a `cn` utility (if present) for Tailwind merging.
+- Forward refs using `React.forwardRef` when exposing focusable elements.
+- Accept `asChild` (if Radix / Slot-like patterns are adopted) for polymorphism—only if pattern already exists; otherwise skip.
+- Expose controlled/uncontrolled patterns (e.g. `value` + `onChange` plus internal state fallback).
+- Avoid implicit side effects; use `useEffect` only for subscriptions / focus trapping / portal mounts.
+
+### Example Skeleton (Button)
+```tsx
+import * as React from 'react';
+import { cn } from '../../utils/cn';
+
+export interface ButtonProps extends React.ButtonHTMLAttributes<HTMLButtonElement> {
+  variant?: 'default' | 'destructive' | 'outline' | 'secondary' | 'ghost' | 'link';
+  size?: 'default' | 'sm' | 'lg' | 'icon';
+  asChild?: boolean;
+}
+
+const Button = React.forwardRef<HTMLButtonElement, ButtonProps>(
+  ({ className, variant = 'default', size = 'default', asChild = false, ...props }, ref) => {
+    const Comp = asChild ? Slot : 'button';
+    
+    return (
+      <Comp
+        className={cn(buttonVariants({ variant, size }), className)}
+        ref={ref}
+        {...props}
+      />
+    );
+  }
+);
+Button.displayName = 'Button';
+
+export { Button };
+```
+
+## Storybook Conventions
+- One `.stories.tsx` per component folder.
+- Showcase: default usage, variants, sizes, interactive state (focus, loading, disabled).
+- Use `play` function (if Storybook Interaction Testing is in use) for basic a11y or keyboard scenarios.
+- Provide `args` with sensible defaults; expose `variant`, `size`, `disabled`, etc.
+- Avoid business data; use placeholder labels.
+
+## Testing Guidelines
+(Adjust to existing test runner—if not present, default to a minimal React Testing Library + Vitest/Jest approach.)
+- Prefer interaction tests: focus management, keyboard activation, ARIA attributes.
+- Snapshot only for stable markup (rare).
+- Test variants produce expected class combinations.
+- Example assertion: `expect(screen.getByRole('button', { name: /submit/i })).toHaveAttribute('data-variant','primary')`.
+
+## Tailwind Usage
+- Keep class lists ordered logically: layout → spacing → typography → color → interaction → state.
+- Abstract repeated class groups into utilities or component variants rather than duplicating.
+- Use responsive + dark variants only when necessary; avoid premature complexity.
+
+## Accessibility Checklist (Copilot should internalize)
+- Every interactive element has an accessible name.
```

Comment:
This should be applied only in cases the regular inferred name is not descriptive enough